packages/foam-vscode/src/features/filtered-resources.ts

```diff
@@ -0,0 +1,252 @@
+import * as fs from 'fs';
+import {
+  IDataStore,
+  URI,
+  FoamWorkspace,
+  Resource,
+  isAttachment,
+  isPlaceholder,
+  getTitle,
+} from 'foam-core';
+import micromatch from 'micromatch';
+import {
+  FilteredResourcesConfig,
+  FilteredResourcesConfigGroupBy,
+} from '../settings';
+import { getContainsTooltip, getNoteTooltip } from '../utils';
+import { dirname, join } from 'path';
+import { FoamFeature } from '../types';
+import { commands } from 'vscode';
+
+const feature: FoamFeature = {
+  activate: (context: vscode.ExtensionContext) => {
+    context.subscriptions.push(
+      commands.registerCommand(
+        'foam-vscode.open-placeholder-note',
+        async (uri: vscode.Uri) => {
+          let dir: string;
+
+          if (vscode.workspace.workspaceFolders) {
+            dir = vscode.workspace.workspaceFolders[0].uri.fsPath.toString();
+          }
+
+          if (!dir) {
+            const activeFile = vscode.window.activeTextEditor?.document;
+            dir = activeFile ? dirname(activeFile.uri.fsPath) : null;
+          }
+
+          if (dir) {
+            const path = join(dir, `${uri.path}.md`);
+            await fs.promises.writeFile(path, `# ${uri.path}`);
+            const ur = vscode.Uri.file(path);
+            await vscode.window.showTextDocument(ur, {
+              preserveFocus: false,
+              preview: false,
+            });
+          }
+        }
+      )
+    );
+  },
+};
+
+export default feature;
+
+export class FilteredResourcesProvider
+  implements vscode.TreeDataProvider<FilteredResourceTreeItem> {
+  // prettier-ignore
+  private _onDidChangeTreeData: vscode.EventEmitter<FilteredResourceTreeItem | undefined | void> = new vscode.EventEmitter<FilteredResourceTreeItem | undefined | void>();
+  // prettier-ignore
+  readonly onDidChangeTreeData: vscode.Event<FilteredResourceTreeItem | undefined | void> = this._onDidChangeTreeData.event;
+  // prettier-ignore
+  private groupBy: FilteredResourcesConfigGroupBy = FilteredResourcesConfigGroupBy.Folder;
+  private exclude: string[] = [];
+  private filteredResources: Resource[] = [];
+  private root = vscode.workspace.workspaceFolders[0].uri.path;
+
+  constructor(
+    private workspace: FoamWorkspace,
+    private dataStore: IDataStore,
+    private filteredNoteType: string,
+    private filteredNoteContextValue: string,
+    private filterPredicate: (resource: Resource, index: number) => boolean,
+    config: FilteredResourcesProviderConfig
+  ) {
+    this.groupBy = config.groupBy;
+    this.exclude = this.getGlobs(config.workspacesURIs, config.exclude);
+    this.setContext();
+    this.computeFilteredResources();
+  }
+
+  setGroupBy(groupBy: FilteredResourcesConfigGroupBy): void {
+    this.groupBy = groupBy;
+    this.setContext();
+    this.refresh();
+  }
+
+  private setContext(): void {
+    vscode.commands.executeCommand(
+      'setContext',
+      `foam-vscode.${this.filteredNoteType}-grouped-by-folder`,
```

Comment:
give that this provider requires specific commands, I believe it should also create/export these commands, e.g. by making them accessible via a `.commands` property, which would be used by the client to register them (there is still the issue of those commands needing to be consistent with what is in `package.json`, but at least we go from 3 places to keep consistent to 2)

I am referring to the commands defined in `features/placeholders.ts:36` (and corresponding ones for orphans)

Reply:
Yes, I like this a lot. I defined the commands as a getter on the provider, and then use a deconstructed array to insert those into the subscription.